File: server/src/models/InventoryItem.js
const mongoose = require('mongoose');

const inventoryItemSchema = new mongoose.Schema({
  name: {
    type: String,
    required: [true, 'Please add an item name'],
    trim: true,
  },
  category: {
    type: String,
    required: [true, 'Please add a category'],
    enum: [
      'Dairy',
      'Vegetables',
      'Fruits',
      'Meat & Poultry',
      'Seafood',
      'Bakery',
      'Pantry',
      'Beverages',
      'Frozen',
      'Others',
    ],
    default: 'Others',
  },
  quantity: {
    type: Number,
    required: [true, 'Please add a quantity'],
    default: 1,
  },
  unit: {
    type: String,
    default: 'pcs',
  },
  expiryDate: {
    type: Date,
    required: [true, 'Please add an expiry date'],
  },
  purchaseDate: {
    type: Date,
    default: Date.now,
  },
  owner: {
    type: mongoose.Schema.ObjectId,
    ref: 'User',
    required: true,
  },
  location: {
    type: String,
    enum: ['Fridge', 'Pantry', 'Freezer'],
    default: 'Fridge',
  },
  status: {
    type: String,
    enum: ['active', 'consumed', 'wasted'],
    default: 'active',
  },
  price: {
    type: Number,
    default: 0,
  },
  imageUrl: {
    type: String,
    default: '',
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

// Index for faster queries on owner and expiry
inventoryItemSchema.index({ owner: 1, expiryDate: 1 });

module.exports = mongoose.model('InventoryItem', inventoryItemSchema);

File: client/lib/features/inventory/domain/inventory_item.dart
class InventoryItem {
  final String id;
  final String name;
  final String category;
  final double quantity;
  final String unit;
  final DateTime expiryDate;
  final DateTime purchaseDate;
  final String location;
  final String status;
  final double price;
  final String imageUrl;

  InventoryItem({
    required this.id,
    required this.name,
    required this.category,
    required this.quantity,
    required this.unit,
    required this.expiryDate,
    required this.purchaseDate,
    required this.location,
    required this.status,
    required this.price,
    required this.imageUrl,
  });

  factory InventoryItem.fromJson(Map<String, dynamic> json) {
    return InventoryItem(
      id: json['_id'],
      name: json['name'],
      category: json['category'],
      quantity: (json['quantity'] as num).toDouble(),
      unit: json['unit'],
      expiryDate: DateTime.parse(json['expiryDate']),
      purchaseDate: DateTime.parse(json['purchaseDate']),
      location: json['location'],
      status: json['status'],
      price: (json['price'] as num).toDouble(),
      imageUrl: json['imageUrl'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'category': category,
      'quantity': quantity,
      'unit': unit,
      'expiryDate': expiryDate.toIso8601String(),
      'purchaseDate': purchaseDate.toIso8601String(),
      'location': location,
      'status': status,
      'price': price,
      'imageUrl': imageUrl,
    };
  }
}
